encoding/_yaml/type/function.ts

```diff
@@ -0,0 +1,41 @@
+// Ported and adapted from js-yaml-js-types v1.0.0:
+// https://github.com/nodeca/js-yaml-js-types/tree/ac537e7bbdd3c2cbbd9882ca3919c520c2dc022b
+// Copyright 2011-2015 by Vitaly Puzrin. All rights reserved. MIT license.
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+import { Type } from "../type.ts";
+import type { Any } from "../utils.ts";
+
+// Note: original implementation used Esprima to handle functions
+// To avoid depencies, we'll just try to check if we can construct a function from given string
+function reconstructFunction(code: string) {
+  const func = new Function(`return ${code}`)();
+  if (!(func instanceof Function)) {
+    throw new TypeError(`Expected function but got ${typeof func}`);
```

Comment:
```suggestion
    throw new TypeError(`Expected function but got ${typeof func}: ${code}`);
```

It would be more user friendly if the error message included the given code.

Reply:
Yes, I applied suggestion 👍🏼